tests/python/pants_test/init/test_plugin_resolver.py

```diff
@@ -115,6 +118,34 @@ def test_exact_requirements(self, unused_test_name, packager_cls):
 
         self.assertEqual(working_set.entries, working_set2.entries)
 
+  @parameterized.expand(INSTALLERS)
+  @skip_unless_python36_and_python37_present
+  def test_exact_requirements_interpreter_change(self, unused_test_name, packager_cls):
+    python36 = PythonInterpreter.from_binary(python_interpreter_path(PY_36))
+    python37 = PythonInterpreter.from_binary(python_interpreter_path(PY_37))
+
+    with self.plugin_resolution(interpreter=python36,
+                                plugins=[('jake', '1.2.3'), ('jane', '3.4.5')],
+                                packager_cls=packager_cls) as results:
+      working_set, chroot, repo_dir, cache_dir = results
+
+      self.assertEqual(2, len(working_set.entries))
+
+      safe_rmtree(repo_dir)
+      with self.assertRaises(Unsatisfiable):
+        with self.plugin_resolution(interpreter=python37,
+                                    chroot=chroot,
+                                    plugins=[('jake', '1.2.3'), ('jane', '3.4.5')]):
+          self.fail('Plugin re-resolution is expected for an incompatible interpreter and it is '
+                    'expected to fail since we removed the dist `repo_dir` above.')
+
+      with self.plugin_resolution(interpreter=python36,
```

Comment:
I recommend moving this above the py37 case and attaching a comment like:

```python
# Even with a deleted repo_dir, this should still work because the interpreter version has not changed.
```

Reply:
That's exactly what the test above does: https://github.com/pantsbuild/pants/pull/8005/files/30706e2a8459e7a45b767523eac0eefc6f1fdf79#diff-d996ef17ea0368099997c262eeb24516R104

If you feel strongly that I should repeat this here I can, but this test builds on that test.

Reply:
OK - I am not a fan of over-explaining, but that characterization is in the eye of the beholder. I added a comment so that you can read just this one test and get it without reading the test above.